sdk/communication/Azure.Communication.Administration/src/PhoneNumberSearchPurchaseOperation.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Communication.Administration.Models;
+
+namespace Azure.Communication.Administration
+{
+    /// <summary>
+    /// Represent a long-running phone number search purchase operation.
+    /// </summary>
+    public class PhoneNumberSearchPurchaseOperation : Operation<PhoneNumberSearch>
+    {
+        private PhoneNumberSearchOperation _searchOperation;
+
+        /// <summary>
+        /// Initializes a new <see cref="PhoneNumberSearchPurchaseOperation"/> instance
+        /// </summary>
+        /// <param name="client">The client used to check for completion.</param>
+        /// <param name="phoneNumberSearchId">The search ID of this operation.</param>
+        /// <param name="initialResponse">The original server response on start operation request.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        internal PhoneNumberSearchPurchaseOperation(
+            PhoneNumberAdministrationClient client,
+            string phoneNumberSearchId,
+            Response initialResponse,
+            CancellationToken cancellationToken = default)
+        {
+            var terminateStatuses = new SearchStatus []
+            {
+                SearchStatus.Success,
+                SearchStatus.Expired,
+                SearchStatus.Cancelled,
+                SearchStatus.Error
+            };
+            _searchOperation = new PhoneNumberSearchOperation(client, phoneNumberSearchId, initialResponse, terminateStatuses, cancellationToken);
```

Comment:
Ah, you're trying to reuse the operation. That introduces an odd dependency. I'd suggest to either extract a base `PhoneNumberOperation<T>` or just extract the common code into static helpers if it's worth it.

Reply:
The dependency is an implementation detail, so instead of having a base class, I had suggested to use a composition and make the helper type internal.

Reply:
I don't mind composition but this particular way feels odd because we're reusing a thing but then override its behavior by specifying the terminal states.

It's a little bit like the PhoneNumberSearchOperation is first class with default statuses, and the other LRO which reuses it is only second class. I'd rather extract the common code and then have both LROs be at the same level.

More like A -> C_internal, B -> C_internal than A, B->A